websites/T/TruckersMP/presence.ts

```diff
@@ -1,380 +1,149 @@
+        { label: "View Forum Category", url: document.URL }
+      ];
+    } else if (document.location.pathname.includes("/topic/")) {
+      presenceData.details = "Viewing a forum topic";
+      presenceData.state = document.querySelector(
+        "main > div > div > div > div > div > div > h1 > span.ipsType_break > span"
+      ).innerHTML;
+      presenceData.buttons = [{ label: "Read Forum Topic", url: document.URL }];
+    } else if (document.location.pathname.includes("/profile/")) {
+      presenceData.details = "Viewing a forum user";
+      presenceData.state = document
+        .querySelectorAll(
+          "main > div > div > div> div > header > div  > div > div"
+        )[1]
+        .querySelector("div > h1 > span").innerHTML;
+      presenceData.buttons = [{ label: "View Forum User", url: document.URL }];
+    }
+    presenceData.details = "Viewing the forum";
+  } else if (document.location.hostname == "map.truckersmp.com") {
+    presenceData.details = "Viewing the TruckersMP Map";
+    if (document.location.pathname.includes("?follow=")) {
+      presenceData.details = "TruckersMP Map";
+      presenceData.state = `Following ${
+        document.querySelector("div.player-name").innerHTML
+      }`;
     }
   } else if (document.location.hostname == "stats.truckersmp.com") {
-    if (document.location.pathname.includes("/history")) {
-      presenceData.details = "TMP Stats, viewing:";
-      presenceData.state = "version history";
-
-      delete presenceData.smallImageKey;
-
-      presence.setActivity(presenceData);
+    presenceData.smallImageKey = "reading";
+    if (document.location.pathname == "/") {
+      presenceData.details = "Viewing the statistics";
     } else if (document.location.pathname.includes("/api")) {
-      presenceData.details = "Reading the API Docs";
-      delete presenceData.state;
-
-      presenceData.smallImageKey = "reading";
-
-      presence.setActivity(presenceData);
+      presenceData.details = "Viewing the";
+      presenceData.state = "API Documentation";
     } else if (document.location.pathname.includes("/live")) {
-      presenceData.details = "Viewing TMP Live Stats";
-      delete presenceData.state;
-      delete presenceData.smallImageKey;
-
-      presence.setActivity(presenceData);
-    } else {
-      presenceData.details = "Viewing TMP Stats";
-      delete presenceData.state;
+      presenceData.details = "Viewing the";
+      presenceData.state = "Live statistics";
+    } else if (document.location.pathname.includes("/history")) {
+      presenceData.details = "Viewing the history";
+    } else if (document.location.pathname.includes("/settings")) {
+      presenceData.details = "Viewing their settings";
       delete presenceData.smallImageKey;
-
-      presence.setActivity(presenceData);
+      presenceData.smallImageKey = "editing";
     }
-  } else {
-    presence.setActivity();
-    presence.setTrayTitle();
   }
+  if (buttons) delete presenceData.buttons;
```

Comment:
Why you delete buttons when it's true or is a positive value

Reply:
yea this doesnt seem right